compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -860,8 +860,9 @@ public BType setLiteralValueAndGetType(BLangLiteral literalExpr, BType expType,
                 }
             }
         }
-
-        if (Types.getReferredType(literalExpr.getBType()).tag == TypeTags.BYTE_ARRAY) {
+        BType referedType = Types.getReferredType(literalExpr.getBType());
+        if (referedType.tag == TypeTags.BYTE_ARRAY ||
+                (referedType.tag == TypeTags.ARRAY && ((BArrayType) referedType).eType.tag == TypeTags.BYTE)) {
             // check whether this is a byte array
```

Comment:
```io:println(table [{ id:1, byteArray: base16 `abcd`}]);``` for an example like this, the array literal ```base16 `abcd` ``` comes two times to this place. The first time, the type of that literal is not set, the correct type is set at this point. But the second time, the type is set to literal properly. Therefore it is not correct to go inside this if-statement and create a new type again.

Reply:
@KavinduZoysa Apparently `io:println(table [{ id:1, byteArray: base16 `abcd`}]);` gives me a compile time error. It should be a bug, isn't it?